classes/Hook/HookActionCarrierProcess.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * 2007-2020 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\Ps_Googleanalytics\Hooks;
+
+use Context;
+use PrestaShop\Module\Ps_Googleanalytics\Handler\GanalyticsDataHandler;
+use PrestaShop\Module\Ps_Googleanalytics\Repository\CarrierRepository;
+use Ps_Googleanalytics;
+
+class HookActionCarrierProcess implements HookInterface
+{
+    private $module;
+    private $context;
+    private $params;
+
+    public function __construct(Ps_Googleanalytics $module, Context $context)
+    {
+        $this->module = $module;
+        $this->context = $context;
+    }
+
+    /**
+     * run
+     *
+     * @return void
+     */
+    public function run()
+    {
+        if (isset($this->params['cart']->id_carrier)) {
+            $carrierRepository = new CarrierRepository();
+            $ganalyticsDataHandler = new GanalyticsDataHandler(
+                $this->context->cart->id,
+                $this->context->shop->id
+            );
+
+            $carrierName = $carrierRepository->findByCarrierId((int) $this->params['cart']->id_carrier);
+            $ganalyticsDataHandler->manageData('MBG.addCheckoutOption(2,\'' . $carrierName . '\');', 'A');
```

Comment:
What is this hardcoded 2 ? 🤔 what does it mean ?

It could be easier to understand if 2 was a `const` 🤔

Reply:
Unfortunately there is no explanation in initial module... I'm not able to tell what the `2` means